client/src/components/ExpandedRowDataEdit.tsx

```diff
@@ -0,0 +1,94 @@
+import React, {
+    MouseEvent,
+    useRef,
+    useState
+} from 'react';
+import {DataEditProps} from '../lib/types';
+import {fetchData} from '../services/fetch';
+import {logError} from '../services/log';
+
+const ExpandedRowDataEdit = ({data, onError, onUpdate}: DataEditProps) => {
+    // This reference object is use to collect updated data by individual form input control.
+    const collectedData = useRef({...data});
+
+    // Use this state to track and temporarily disabled all buttons when Save button is clicked.
+    const [saveClicked, setSaveClicked] = useState(false);
```

Comment:
Not a blocker, but it might be worth renaming this state to something like 

```suggestion
    const [isSaving, setIsSaving] = useState(false);
```